src/transformers/models/rt_detr/image_processing_rt_detr_fast.py

```diff
@@ -0,0 +1,856 @@
+            else:
+                interpolation = resample
+            resized_images = [self.resize(image, size=size, interpolation=interpolation) for image in images]
+            if annotations is not None:
+                for i, (image, target) in enumerate(zip(resized_images, annotations)):
+                    annotations[i] = self.resize_annotation(
+                        target,
+                        orig_size=images[i].size()[-2:],
+                        target_size=image.size()[-2:],
+                    )
+            images = resized_images
+            del resized_images
+
+        if do_rescale and do_normalize:
+            # fused rescale and normalize
+            new_mean = torch.tensor(image_mean, device=images[0].device) * (1.0 / rescale_factor)
+            new_std = torch.tensor(image_std, device=images[0].device) * (1.0 / rescale_factor)
+            images = [F.normalize(image.to(dtype=torch.float32), new_mean, new_std) for image in images]
+        elif do_rescale:
+            images = [image * rescale_factor for image in images]
+        elif do_normalize:
+            images = [F.normalize(image, image_mean, image_std) for image in images]
+
+        if do_convert_annotations and annotations is not None:
+            annotations = [
+                self.normalize_annotation(annotation, get_image_size(image, input_data_format))
+                for annotation, image in zip(annotations, images)
+            ]
+
+        if do_pad:
+            # Pads images and returns their mask: {'pixel_values': ..., 'pixel_mask': ...}
+            if pad_size is not None:
+                padded_size = (pad_size["height"], pad_size["width"])
+            else:
+                padded_size = get_max_height_width(images)
+
+            annotation_list = annotations if annotations is not None else [None] * len(images)
+            padded_images = []
+            pixel_masks = []
+            padded_annotations = []
+            for image, annotation in zip(images, annotation_list):
+                if padded_size == image.size()[-2:]:
+                    padded_images.append(image)
+                    pixel_masks.append(torch.ones(padded_size, dtype=torch.int64, device=image.device))
+                    padded_annotations.append(annotation)
+                    continue
+                padded_image, pixel_mask, padded_annotation = self.pad(
+                    image, padded_size, annotation=annotation, update_bboxes=do_convert_annotations
+                )
+                padded_images.append(padded_image)
+                pixel_masks.append(pixel_mask)
+                padded_annotations.append(padded_annotation)
+            images = padded_images
+            if annotations is not None:
+                annotations = padded_annotations
+            del padded_images, padded_annotations
+            data.update({"pixel_mask": torch.stack(pixel_masks, dim=0)})
+
```

Comment:
there is again the recurring theme of iterating on the images a bunch of time. Now I don't know if you can avoid it, but if you can let's only iterate through them once!